app/controllers/users_controller.rb

```diff
@@ -52,7 +52,7 @@ def edit
 
   def registration_code
     begin
-      flash[:notice] = User.change_registration_code(params[:registration_code])
+      flash[:notice] = User.change_code("regular", params[:registration_code])
```

Comment:
Good, thank you for refactoring and removing that redundancy between regular and admin codes.